src/components/BlogForm/NewPost.tsx

```diff
@@ -0,0 +1,117 @@
+"use client"; // needed for useform
+import { Button, Stack, TextField, Typography } from "@mui/material";
+import { useForm, Controller } from "react-hook-form";
+import { postBlog } from "../../api/blog";
+import { useSession } from "next-auth/react";
+import { DataProps } from "./types";
+
+const NewPost = () => {
+  const { data: session } = useSession();
+  const user = session?.user.name as string;
```

Comment:
move this line to before the return and if (!session?.user.name) return <LoadingScreen />; and then handle variable assignment afterward without needing null or type check